File: palindrome_num.py
def isPalindrome(x: int) -> bool:
    if x < 0:
        return False
    
    rev = 0
    temp = x
    
    while temp > 0:
        rev = rev * 10 + temp % 10
        temp //= 10  
    
    return x == rev

if __name__ == "__main__":
    x = int(input("Enter number: "))
    
    if isPalindrome(x):
        print("Palindrome")
    else:
        print("Not Palindrome")

File: Day-3/palindrome_num.cpp
#include <iostream>
using namespace std;
bool isPalindrome(long long int x) {
        if (x < 0) return false;
        long long int rev = 0, temp = x;
        while (temp > 0) {
            rev = rev * 10 + temp % 10;
            temp /= 10;
        }
        return x == rev;
    }

int main() {

    long long int x;
    cout << "Enter number: ";
    cin >> x;

    if (isPalindrome(x)) {
        cout << "Palindrome";
    } else {
        cout << "Not Palindrome";
    }

    return 0;
}